test/integration/node-specific/bson-options/useBigInt64.test.ts

```diff
@@ -0,0 +1,125 @@
+import { expect } from 'chai';
+
+import { Collection, Db, MongoClient } from '../../../mongodb';
+import { setupDatabase } from '../../shared.js';
+
+describe('useBigInt64 option', function () {
+  let configuration;
+  beforeEach(function () {
+    configuration = this.configuration;
+    return setupDatabase(this.configuration);
+  });
+
+  describe('when not provided to client', async function () {
+    let client: MongoClient;
+
+    beforeEach(async function () {
+      client = await configuration.newClient(configuration.writeConcernMax());
+    });
+
+    afterEach(async function () {
+      await client.close();
+    });
+
+    it('is set to driver default (useBigInt64=false)', async function () {
+      expect(client.s.bsonOptions.useBigInt64).to.exist;
+      expect(client.s.bsonOptions.useBigInt64).to.be.false;
+    });
+  });
+
+  describe('when set at client level', function () {
+    let client: MongoClient;
+    beforeEach(async function () {
+      client = configuration.newClient(configuration.writeConcernMax(), { useBigInt64: true });
+    });
+
+    afterEach(async function () {
+      await client.close();
+    });
+
+    it('supercedes driver level', function () {
+      expect(client.s.bsonOptions.useBigInt64).to.exist;
+      expect(client.s.bsonOptions.useBigInt64).to.be.true;
+    });
+  });
+
```

Comment:
Can we add a test in this file as well to check the error case `useBigInt64` set to `true` and `promoteLongs` is `false`. The patch linked shows that happening but since it's called out in the acceptance criteria we should explicitly test it here.

Reply:
Added the new test, but the options are only checked on the call to `MongoClient.connect`. Doing the check implicitly in connect because it has to use BSON.deserialize does satisfy the AC, but feels brittle.

Do we want to explicitly do this check in the constructor for the `MongoClient`? It could be done in the `parseOptions` helper function in src/connection_string.ts where we validate the other options.

Reply:
Doing it in `parseOptions` sounds like a good idea.